site.py

```diff
@@ -315,11 +320,13 @@ def check_container(cid):
 # Updated using the function above.
 @app.route('/containers/<uuid>/status', methods=['GET'])
 def container_status(uuid):
-    cid = container_id(uuid)
-    url = '/check_container/' + cid[:16]
+    url = launch_container(uuid)
```

Comment:
Although not obvious, this is a clever solution that bypasses the need to use a separate thread. For this to work reliably, the folder named after the [`cid`](https://github.com/swanijam/HaaS/blob/status-and-logs-fix/site.py#L63) should be checked for before this line to ensure the files exist for the generated UUID. Something like:

```python
folder = os.path.join(CONTAINER_DIR, uuid)
if not os.path.exists(folder):
    return '', 404
```

In which case the we page polling this status should display a friendly *No container exists* message.